pydrive2/fs/spec.py

```diff
@@ -49,13 +55,120 @@ def should_retry(exc):
     )(func)
 
 
+def _cache_file(client_id):
+    cache_dir = appdirs.user_cache_dir("pydrive2fs", appauthor=False)
+    os.makedirs(cache_dir, exist_ok=True)
+    return os.path.join(cache_dir, client_id + ".json")
+
+
 class GDriveFileSystem(AbstractFileSystem):
-    def __init__(self, path, google_auth, trash_only=True, **kwargs):
+    GDRIVE_CREDENTIALS_DATA = "GDRIVE_CREDENTIALS_DATA"
+
+    def __init__(
+        self,
+        path,
+        google_auth=None,
+        trash_only=True,
+        client_id=None,
+        client_secret=None,
+        client_user_email=None,
+        client_json_file_path=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
         self.path = path
         self.root, self.base = self.split_path(self.path)
+
+        if not google_auth:
+            google_auth = self._auth(
+                client_id=client_id,
+                client_secret=client_secret,
+                client_user_email=client_user_email,
+                client_json_file_path=client_json_file_path,
+            )
+
         self.client = GoogleDrive(google_auth)
         self._trash_only = trash_only
-        super().__init__(**kwargs)
+
+    def _auth(
+        self,
+        client_id=None,
+        client_secret=None,
+        client_user_email=None,
+        client_json_file_path=None,
+    ):
+        settings = {
+            "get_refresh_token": True,
+            "oauth_scope": [
+                "https://www.googleapis.com/auth/drive",
+                "https://www.googleapis.com/auth/drive.appdata",
+            ],
+        }
+
+        env_creds = os.getenv(self.GDRIVE_CREDENTIALS_DATA)
+
+        if (
+            not env_creds
+            and not client_json_file_path
+            and not (client_id and client_secret)
+        ):
+            raise ValueError(
+                "Specify credentials using one of these methods: "
+                "client_id/client_secret or "
+                "client_json_file_path or "
+                f"{self.GDRIVE_CREDENTIALS_DATA}"
+            )
+
+        if env_creds:
+            settings["save_credentials"] = True
+            settings["save_credentials_backend"] = "dictionary"
+            settings["save_credentials_dict"] = {"creds": env_creds}
+            settings["save_credentials_key"] = "creds"
+        elif not client_json_file_path:
+            settings["save_credentials"] = True
+            settings["save_credentials_backend"] = "file"
+            settings["save_credentials_file"] = _cache_file(client_id)
```

Comment:
it's quite limiting unfortunately. It's quite handy and easy to use the same client/secret and use different Google accounts now in different projects, should we introduce a profile name? (by default it can be empty).

overriding this? e.g. to let ppl create global file per machine (like you can do in DVC) or put them close to the project, etc ...